### DISCUSSION TITLE
New `app` directory 404 page issue/question

### DISCUSSION THREAD

**Discussion by TheoKondak** [vercel/community#2295] [Help] [answered]
### Summary

Recently I have updated my portfolio to the latest Nextjs 13, and started adopting the new `app/` directory approach for routing. 

The app is quite simple for now, having a home page, and then some dynamically generated pages under `experiences/[experience]`, which is a list of work experience articles. 

Now I am creating the `not-found` page, using one `not-found.tsx` file in the `app/`.

When I type a non-existing path, the `not-found`  page is displayed, but there is an issue. The app, keeps requesting the non-existing page every three seconds or so, resulting in console errors.

![image](https://user-images.githubusercontent.com/25774466/236170893-65f74107-8bc0-41c1-986c-188dac4c2201.png)

I have searched the beta docs, but could not understand what I am missing. I probably have to write some middleware but at this point, I am not sure how to do that.

Thank you in advance!

### Example

https://github.com/TheoKondak/portolio-minimal-reproduction

**Comment by amyegan** [comment]
Thanks for reaching out, @TheoKondak! I reached out to the Next.js team, and they asked that you open a new issue in the [next.js repo](https://github.com/vercel/next.js/issues) to help them keep track of it while they investigate 🙏

**Reply by TheoKondak** [reply]
Hi @amyegan . Thanks for trying to help.

 I have submitted an [issue](https://github.com/vercel/next.js/issues/49281), but I don't think it's an issue with NextJs. I believe that it is something wrong on my part.

**Comment by TheoKondak** [comment, marked as answer]
Solution proposed by [@OatsProgramming](https://github.com/OatsProgramming) in the [issue](https://github.com/vercel/next.js/issues/49281#issuecomment-1537070336).

To make it easier for anyone else in the same situation, I will summarize what I did following your approach and solve the issue:

- In the root of `app/` create a new folder and give it a name of `[...not-found]`. The name of the folder is not important, but it's better to call it `not-found` for consistency.
- Create a new `page.tsx` file. The content of the file:

```js
    import { notFound } from 'next/navigation';

     export default function NotFound() {
           notFound();
      }
```

This means that whenever a route does not match with an existing route in your project, it will call the newly created page, and then call the `notFound()` which will redirect to the `not-found.tsx` page.
